src/App.js

```diff
@@ -1,75 +1,134 @@
 import React from 'react';
+import TodoList from './TodoList';
+import Footer from './Footer';
 
-function App() {
-  return (
-    <section className="todoapp">
-      <header className="header">
-        <h1>todos</h1>
+const FILTER_TYPES = {
+  all: 'All',
+  active: 'Active',
+  completed: 'Completed',
+};
 
-        <input
-          className="new-todo"
-          placeholder="What needs to be done?"
-        />
-      </header>
-
-      <section className="main" style={{ display: 'block' }}>
-        <input type="checkbox" id="toggle-all" className="toggle-all" />
-        <label htmlFor="toggle-all">Mark all as complete</label>
-
-        <ul className="todo-list">
-          <li className="">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-1" />
-              <label htmlFor="todo-1">sdfsdfsdf</label>
-              <button type="button" className="destroy" />
-            </div>
-          </li>
-
-          <li className="">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-2" />
-              <label htmlFor="todo-2">sakgjdfgkhjasgdhjfhs</label>
-              <button type="button" className="destroy" />
-            </div>
-          </li>
-
-          <li className="">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-3" />
-              <label htmlFor="todo-3">sddfgdfgdf</label>
-              <button type="button" className="destroy" />
-            </div>
-          </li>
-        </ul>
-      </section>
+class App extends React.Component {
+  state = {
+    inputValue: '',
+    todos: [],
+    counterId: 1,
+    selectedFilter: FILTER_TYPES.all,
+  }
+
+  Id = () => {
+    const { counterId } = this.state;
+
+    this.setState(state => ({
+      counterId: state.counterId + 1,
+    }));
+
+    return counterId;
+  }
+
+  hundleInputValue = (event) => {
+    this.setState({
+      inputValue: event.target.value,
+    });
+  }
+
+  onSubmit = (event) => {
+    event.preventDefault();
+    this.setState(state => ({
```

Comment:
```suggestion
    this.setState(prevState => ({
```
в этом объекте хранится предыдущее состояние state